apps/framework-cli/src/infrastructure/redis/redis_client.rs

```diff
@@ -197,6 +199,14 @@ impl RedisClient {
         if self.is_leader {
             info!("Instance {} became leader", self.instance_id);
             self.renew_lock().await?;
+
+            // Call the leadership callback if set
+            if let Some(callback) = &self.leadership_callback {
+                let callback = callback.clone();
+                tokio::spawn(async move {
```

Comment:
Spawn is not strictly necessary here. - but does ensure non-blocking concurrency.  A consideration is that from a library/module perspective, it is unsure how long a created callback might be and, thus, how long it would otherwise block.